module_info_import/models/module_information.py

```diff
@@ -13,95 +14,84 @@
-            )
-            raise UserError(ERROR_MESSAGE)
-        return data
+    def synchronize_module(self):
+        versions = self.env["odoo.version"].search([])
+        for version in versions:
+            data = self.get_module_info(version.name)
+            for orga, repos in data.items():
+                for repo, modules in repos.items():
+                    for module in modules:
+                        self._update_or_create_modules(version, orga, repo, module)
 
     @api.model
-    def _get_module_information_vals_from_github_stat(self, module_info):
-        return {
-            "name": module_info["name"],
-            "description_rst": module_info["description_rst"],
-            "description_html": module_info["description_rst_html"],
-            "authors": module_info["author_ids_description"],
+    def _update_or_create_modules(self, version, orga_name, repo_name, module):
+        repo = self._get_or_create_repo(orga_name, repo_name)
+        vals = {
+            "repo_id": repo.id,
+            "technical_name": module,
+            # "description_rst":info["description"]
+            # # TODO add description in odoo-module-tracker
         }
+        module = self.search(
+            [("technical_name", "=", module), ("partner_id", "=", False)]
+        )
+        if module:
+            if module._should_update_module(version.name, orga_name):
+                module.write(vals)
+            module._add_available_version(version)
+        else:
+            vals.update({"available_version_ids": [(4, version.id, 0)]})
+            _logger.info(">>>> CREATE MODULE : %s", vals)
+            module = self.create(vals)
 
     @api.model
-    def _update_modules(self, data):
-        # Since the goal is to resynchronize all modules and versions, it
-        # seems appropriate to make one big search and then work with cache.
-        #        existing_module = self.search([])
-        #        test = dict([(x.name, x.id) for x in existing_module])
-        #        existing_module_names = test.keys()
-        #        for module_info in data:
-        #            if module_info['technical_name'] in existing_module_names:
-        #                module_id = test.get(module_info['technical_name'])
-        #                self.browse(module_id).write({'name': module_info['technical_name']})
-        #            else:
-        #                self.create({'name': {'name': module_info['technical_name']}})
-        odoo_version_dct = dict(
-            [(v.version, v.id) for v in self.env["odoo.version"].search([])]
+    def _get_or_create_repo(self, orga_name, repo_name):
+        repo_vals = {"organization": orga_name, "name": repo_name}
+        repo = self.env["module.repo"].search(
+            [("name", "=", repo_name), ("organization", "=", orga_name)]
         )
+        if not repo:
+            repo = self.env["module.repo"].create(repo_vals)
+        return repo
```

Comment:
It's weird to define the variable  `repo_vals` even if we do not need it.
Maybe something like that will be more readable

```suggestion
    def _get_or_create_repo(self, orga_name, repo_name):
        repo = self.env["module.repo"].search(
            [("name", "=", repo_name), ("organization", "=", orga_name)]
        )
        if repo:
            return repo
        else:
            return self.env["module.repo"].create({
                "organization": orga_name,
                "name": repo_name
                })
```